unittests/test_sanity_functions.py

```diff
@@ -213,247 +214,239 @@ def setUp(self):
-                               sn.evaluate, sn.assert_eq(1, 2))
-        self.assertRaisesRegex(SanityError, '1 != False',
-                               sn.evaluate, sn.assert_eq(1, False))
-        self.assertRaisesRegex(
-            SanityError, '1 is not equals to 2',
-            sn.evaluate, sn.assert_eq(1, 2, '{0} is not equals to {1}'))
+        assert sn.assert_eq(1, 1)
+        assert sn.assert_eq(1, True)
+        with pytest.raises(SanityError, match='1 != 2'):
+            sn.evaluate(sn.assert_eq(1, 2))
+        with pytest.raises(SanityError, match='1 != False'):
+            sn.evaluate(sn.assert_eq(1, False))
+        with pytest.raises(SanityError, match='1 is not equals to 2'):
+            sn.evaluate(sn.assert_eq(1, 2, '{0} is not equals to {1}'))
 
     def test_assert_eq_with_deferrables(self):
-        self.assertTrue(sn.assert_eq(1, sn.defer(1)))
-        self.assertTrue(sn.assert_eq(sn.defer(1), True))
-        self.assertRaisesRegex(SanityError, '1 != 2',
-                               sn.evaluate, sn.assert_eq(sn.defer(1), 2))
-        self.assertRaisesRegex(SanityError, '1 != False',
-                               sn.evaluate, sn.assert_eq(sn.defer(1), False))
+        assert sn.assert_eq(1, sn.defer(1))
+        assert sn.assert_eq(sn.defer(1), True)
+        with pytest.raises(SanityError, match='1 != 2'):
+            sn.evaluate(sn.assert_eq(sn.defer(1), 2))
+        with pytest.raises(SanityError, match='1 != False'):
+            sn.evaluate(sn.assert_eq(sn.defer(1), False))
 
     def test_assert_ne(self):
-        self.assertTrue(sn.assert_ne(1, 2))
-        self.assertTrue(sn.assert_ne(1, False))
-        self.assertRaisesRegex(SanityError, '1 == 1',
-                               sn.evaluate, sn.assert_ne(1, 1))
-        self.assertRaisesRegex(SanityError, '1 == True',
-                               sn.evaluate, sn.assert_ne(1, True))
+        assert sn.assert_ne(1, 2)
+        assert sn.assert_ne(1, False)
+        with pytest.raises(SanityError, match='1 == 1'):
+            sn.evaluate(sn.assert_ne(1, 1))
+        with pytest.raises(SanityError, match='1 == True'):
+            sn.evaluate(sn.assert_ne(1, True))
 
     def test_assert_ne_with_deferrables(self):
-        self.assertTrue(sn.assert_ne(1, sn.defer(2)))
-        self.assertTrue(sn.assert_ne(sn.defer(1), False))
-        self.assertRaisesRegex(SanityError, '1 == 1',
-                               sn.evaluate, sn.assert_ne(sn.defer(1), 1))
-        self.assertRaisesRegex(SanityError, '1 == True',
-                               sn.evaluate, sn.assert_ne(sn.defer(1), True))
+        assert sn.assert_ne(1, sn.defer(2))
+        assert sn.assert_ne(sn.defer(1), False)
+        with pytest.raises(SanityError, match='1 == 1'):
+            sn.evaluate(sn.assert_ne(sn.defer(1), 1))
+        with pytest.raises(SanityError, match='1 == True'):
+            sn.evaluate(sn.assert_ne(sn.defer(1), True))
```

Comment:
The file has more `with pytest.raises`. I am not sure if it makes sense to leave a blank line after every one of them or if it will be unnecessarily long

Reply:
Our coding style (though not documented explicitly, which I should do that) prescribes that we should leave a blank line after each block of code. `with` defines a block, so we should leave a blank line afterwards.